packages/medusa/src/services/cache.ts

```diff
@@ -21,7 +21,7 @@ export default class CacheService implements ICacheService {
    */
   async set(
     key: string,
-    data: Record<string, unknown>,
+    data: Record<string, unknown> | Record<string, unknown>[],
```

Comment:
suggestion: In order to not constraint too much the type without knowing what we can pass, we can template it wdyt?

Reply:
I am not sure what you mean by templating it?

Reply:
set<T = unknown>(..., data: T)